src/iop/toneequal.c

```diff
@@ -1725,36 +1725,33 @@ static void smoothing_callback(GtkWidget *slider, gpointer user_data)
 static void auto_adjust_exposure_boost(GtkWidget *quad, gpointer user_data)
 {
   dt_iop_module_t *self = (dt_iop_module_t *)user_data;
+  dt_iop_toneequalizer_params_t *p = (dt_iop_toneequalizer_params_t *)self->params;
+  dt_iop_toneequalizer_gui_data_t *g = (dt_iop_toneequalizer_gui_data_t *)self->gui_data;
+
   if(darktable.gui->reset) return;
 
   dt_iop_request_focus(self);
 
-  if(!self->enabled)
-  {
-    // If module disabled, enable and do nothing
-    dt_dev_add_history_item(darktable.develop, self, TRUE);
-    return;
-  }
-
-  dt_iop_toneequalizer_params_t *p = (dt_iop_toneequalizer_params_t *)self->params;
-  dt_iop_toneequalizer_gui_data_t *g = (dt_iop_toneequalizer_gui_data_t *)self->gui_data;
-
-  if(p->exposure_boost != 0.0f)
+  if(p->exposure_boost != 0.0f || !self->enabled)
   {
     // Reset the exposure boost and do nothing
     p->exposure_boost = 0.0f;
     ++darktable.gui->reset;
     dt_bauhaus_slider_set_soft(g->exposure_boost, p->exposure_boost);
+    dt_bauhaus_widget_set_quad_active(quad, FALSE);
     --darktable.gui->reset;
 
     invalidate_luminance_cache(self);
     dt_dev_add_history_item(darktable.develop, self, TRUE);
-    dt_bauhaus_widget_set_quad_active(quad, FALSE);
     return;
   }
 
-  if(!g->luminance_valid || self->dev->pipe->processing)
+  if(!g->luminance_valid || self->dev->pipe->processing || !g->histogram_valid)
   {
+    ++darktable.gui->reset;
+    gboolean active = dt_bauhaus_widget_get_quad_active(quad);
```

Comment:
const

Reply:
Ok, I will replace the pickers with a regular non-toggle button, possibly with a new icon. Maybe a simple "A" standing for "auto", or something resembling that.